piksi_tools/stm_unique_id.py

```diff
@@ -14,36 +14,51 @@
 import sys
 import serial_link
 
+from piksi_tools.heartbeat import Heartbeat
+
 from sbp.flash          import SBP_MSG_STM_UNIQUE_ID_DEVICE
 from sbp.client.handler import *
 
-class STMUniqueID:
+class STMUniqueID(object):
+  """
+  Retrieve the STM Unique ID from Piksi.
+  """
 
   def __init__(self, link):
-    self.heartbeat_received = False
+    """
+    Parameters
+    ==========
+    link : sbp.client.handler.Handler
+      link to register Heartbeat message callback with
+    """
     self.unique_id_returned = False
-    # SBP version is unset in older devices.
-    self.sbp_version = (0, 0)
     self.unique_id = None
     self.link = link
-    link.add_callback(self.receive_heartbeat, SBP_MSG_HEARTBEAT)
+    self.heartbeat = Heartbeat(link)
     link.add_callback(self.receive_stm_unique_id_callback, SBP_MSG_STM_UNIQUE_ID_DEVICE)
```

Comment:
I'm still failing to see how this heartbeat here is different than the stm callback you have on the next line. You can construct a Hearbeat object implementing `__call__` and have equivalent behavior without having this heartbeat callback be aware of the serial connection. Yeah, you have to remove the callback, but you can just do that like how you're removing the stm callback at 44.